pkg_resources/__init__.py

```diff
@@ -3183,7 +3333,7 @@ def _always_object(classes):
     return classes
 
 
-def _find_adapter(registry, ob):
+def _find_adapter(registry: Mapping[type, _AdapterType], ob: object):
```

Comment:
Here the way to go is probably have 3 definitions using `typing.overload` for the 3 different types of adapters and compatible `ob` (something like `_ProviderFactory`+ `_LoaderProtocol`; `_DistributionFider`+`_Importer`; `_NamespaceHandler`+`_Importer`, with `_Importer` a protocol that is compatible with `PathEntryFinder` and/or `MetaPathFinder`? ... considering the limitations in https://github.com/python/typeshed/issues/11541 and https://github.com/python/typeshed/issues/2468)